api/helpers.go

```diff
@@ -1,17 +1,63 @@
 package api
 
 import (
+	"bytes"
 	"context"
 	"encoding/json"
+	"html/template"
 	"net/http"
 	"time"
 
 	"github.com/go-chi/chi/v5"
 	"github.com/lestrrat-go/jwx/v2/jwt"
 	"github.com/vocdoni/saas-backend/db"
+	"github.com/vocdoni/saas-backend/notifications"
 	"go.vocdoni.io/dvote/log"
 )
 
+// sendNotification method sends a notification to the email provided. It
+// requires the email, the name of the recipient, the subject, the plain body,
+// the mail template and the data to fill the template. It returns an error if
+// the mail service is available and the notification could not be sent. If the
+// mail service is not available, the notification is not sent but the function
+// returns nil.
+func (a *API) sendNotification(ctx context.Context, email, name, subject,
+	plainbody string, temp notifications.MailTemplate, data any,
+) error {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+	defer cancel()
+	// send the verification code via email if the mail service is available
+	if a.mail != nil {
+		ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+		defer cancel()
```

Comment:
why is there a second context created here?

Reply:
you are right, is a copy-paste error